### PR TITLE
Update config.nims search path in docs

### PR DESCRIPTION
**Pull request by kaushalmodi**
Related: https://github.com/nim-lang/Nim/pull/8682

/cc @timotheecour @Araq

### CODE REVIEW

**Comment by kaushalmodi** [comment]
@Araq Looks like the failure is only due to tfragment_gc on OSX: https://travis-ci.org/nim-lang/Nim/jobs/442193457#L3201

The PR touches only the nims.rst file.

File: doc/nims.rst [hunk 1]
```diff
@@ -6,9 +6,25 @@ Strictly speaking, ``NimScript`` is the subset of Nim that can be evaluated
...
+The ``nim`` executable processes the ``.nims`` configuration files in
+the following directories (in this order; later files overwrite
+previous settings):
+
+1) If environment variable ``XDG_CONFIG_HOME`` is defined,
+   ``$XDG_CONFIG_HOME/nim/config.nims`` or
+   ``~/.config/nim/config.nims`` (POSIX) or
+   ``%APPDATA%/nim/config.nims`` (Windows). This file can be skipped
+   with the ``--skipUserCfg`` command line option.
+2) ``$parentDir/config.nims`` where ``$parentDir`` stands for any
```
Comment (Araq) [inline doc/nims.rst]: This hasn't been implemented.

Comment (kaushalmodi) [reply doc/nims.rst]: You mean that the `--skip*` switches don't apply to .nims files? I then misunderstood the above referenced applied PR.

Comment (kaushalmodi) [reply doc/nims.rst]: Is this true then?

```rst
1) If environment variable ``XDG_CONFIG_HOME`` is defined,
   ``$XDG_CONFIG_HOME/nim/config.nims`` or
   ``~/.config/nim/config.nims`` (POSIX) or
   ``%APPDATA%/nim/config.nims`` (Windows) will be used.
2) Next, if ``config.nims`` is placed inside a directory, that will
   apply to all Nim projects inside that directory.
3) Next, if a project has ``<project>.nims`` that resides in the same
   directory as the ``<project>.nim``, that will apply finally.
```

Comment (timotheecour) [reply doc/nims.rst]: @Araq  I believe I did implement it, see https://github.com/nim-lang/Nim/pull/8682/files#diff-5fdd8f1b1ac3aa10d48e0af960fb8c17R52
when I pass `--skipUserCfg`, /Users/timothee/.config/nim/config.nims is skipped

Comment (Araq) [reply doc/nims.rst]: Never mind, I misremembered then.

**Comment by Araq** [comment]
Unrelated test failure, merging.

**Pull request merged**